File: articles/the-future-of-algorithms-computational-reasoning/java/AutomationRisk.java
public class AutomationRisk {
    static double automationRisk(double stakes, double opacity, double delegation, double irreversibility) {
        return Math.max(0.0, Math.min(1.0, stakes * opacity * delegation * irreversibility));
    }

    public static void main(String[] args) {
        System.out.printf("%.6f%n", automationRisk(0.95, 0.85, 0.90, 0.80));
    }
}

File: articles/the-future-of-algorithms-computational-reasoning/typescript/automation_risk.ts
function automationRisk(stakes: number, opacity: number, delegation: number, irreversibility: number): number {
  return Math.max(0, Math.min(1, stakes * opacity * delegation * irreversibility));
}

console.log(automationRisk(0.95, 0.85, 0.90, 0.80).toFixed(6));
